runners/gradle-plugin/build.gradle.kts

```diff
@@ -32,13 +28,6 @@ dependencies {
                 .classpath.asFiles.first()
         )
     )
-
-    constraints {
-        val kotlin_version: String by project
-        compileOnly("org.jetbrains.kotlin:kotlin-reflect:${kotlin_version}") {
-            because("kotlin-gradle-plugin and :core both depend on this")
```

Comment:
This is a [dependency constraint](https://docs.gradle.org/current/userguide/dependency_constraints.html). 
It allows to choose needed version of `kotlin-reflect` from transitive dependencies. Currently it makes no sense because the latest `kotlin-gradle-plugin` excludes `kotlin-reflect`.